File: C_by_Harry/CH-3_practice_set.c/02_problem.c
#include <stdio.h>

int main()
{
    // Write a program to determine whether a student has passed or failed. To pass, a
    //  student requires a total of 40% and at least 33% in each subject. Assume there
    //  are three subjects and take the marks as input from the user.
    int m1, m2, m3;
    // int value_of_marks; //write only vom, then hit enter,vscode automatically makes correct longer name
    printf("Enter marks of 1st subject : \n");
    scanf("%d", &m1);
    printf("Enter marks of 2nd subject : \n");
    scanf("%d", &m2);
    printf("Enter marks of 3rd subject : \n");
    scanf("%d", &m3);

    if (m1 < 33 || m2 < 33 || m3 < 33)
    {
        printf("You are failed due to less than 33 marks in individual subject(s)\n");
    }
    else if ((m1 + m2 + m3) / 3 < 40)
    {
        printf("You are failed as your average marks are less than 40% \n");
    }
    else{
        printf("You are passed\n");
    }
    return 0;
}

File: python_by_harry/CH-6 practice set/02_problem.py
"""
2. Write a program to find out whether a student has passed or failed if it requires a 
total of 40% and at least 33% in each subject to pass. Assume 3 subjects and 
take marks as an input from the user. 

"""
sub_1 = int(input("Enter marks of a subject out of 100 : "))
sub_2 = int(input("Enter marks of a subject out of 100 : "))
sub_3 = int(input("Enter marks of a subject out of 100 : "))

total_percentage = ((sub_1 + sub_2 + sub_3)/300)*100

if (sub_1<0 or sub_1>100 or sub_2<0 or sub_2>100 or sub_3<0 or sub_3>100):
     print("INVALID MARKS")

elif(total_percentage>=40 and (sub_1>=33 and sub_2>=33 and sub_3>=33)):
    print("YOU ARE PASSED!!!")

else:
     print("YOU ARE FAILED!!!")

